bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1997,6 +1997,8 @@ void BCStateTran::processData() {
         Assert(getFetchingState() == FetchingState::GettingMissingResPages);
 
         LOG_DEBUG(STLogger, "moved to GettingMissingResPages");
+        DataStore::CheckpointDesc cp = g.txn()->getCheckpointBeingFetched();
+        replicaForStateTransfer_->onTransferringComplete(cp.checkpointNum);
```

Comment:
I'm not sure it's a right behaviour: onTransferring complete is called only after the last block which is a vBlock

Reply:
Agreed. I think this might cause all sorts of problems. We probably need a different solution to this.

Reply:
You are right. I'm closing this PR.